File: 0344-reverse-string/0344-reverse-string.py
class Solution(object):
    def reverseString(self, s):
        """
        :type s: List[str]
        :rtype: None Do not return anything, modify s in-place instead.
        """
        right = len(s)-1
        left = 0 
        while left<=right:
            temp = s[left]
            s[left] = s[right]
            s[right] = temp
            left +=1
            right -=1
        # def reverse(l,r):
        #     if l < r:
        #         s[l],s[r] = s[r],s[l]
        #         reverse(l+1,r-1)
        # reverse(0,len(s)-1)

File: 0344-reverse-string/0344-reverse-string.js
/**
 * @param {character[]} s
 * @return {void} Do not return anything, modify s in-place instead.
 */
var reverseString = function(s) {
    for(let i = 0;i<Math.trunc(s.length/2);i++){ // Math.trunc ==> removes the decimal part .
        let temp = s[i];
        s[i] = s[s.length-i-1];
        s[s.length-i-1] = temp;
    }
};